dubbo-registry/dubbo-registry-redis/src/main/java/org/apache/dubbo/registry/redis/RedisRegistry.java

```diff
@@ -294,10 +308,12 @@ private void doNotify(Collection<String> keys, URL url, Collection<NotifyListene
             if (CollectionUtils.isNotEmptyMap(values)) {
                 for (Map.Entry<String, String> entry : values.entrySet()) {
                     URL u = URL.valueOf(entry.getKey());
+                    long expire = Long.parseLong(entry.getValue());
                     if (!u.getParameter(DYNAMIC_KEY, true)
-                            || Long.parseLong(entry.getValue()) >= now) {
+                            || expire >= now) {
                         if (UrlUtils.isMatch(url, u)) {
                             urls.add(u);
+                            expireCache.put(u, expire);

```

Comment:
Is it possible for expireCache to go leaking for it's never cleared?

Reply:
ok , add remove logic and related UT

Reply:
Also, I don't see where `expireCache` is used inside `doNotify`.

Reply:
`expireCache` mark which service may be down and call `doNotity` to fetch latest data from redis